modules/reindex/src/internalClusterTest/java/org/elasticsearch/index/reindex/ReindexPluginMetricsIT.java

```diff
@@ -0,0 +1,263 @@
+import static org.elasticsearch.reindex.ReindexMetrics.REINDEX_TIME_HISTOGRAM;
+import static org.elasticsearch.reindex.UpdateByQueryMetrics.UPDATE_BY_QUERY_TIME_HISTOGRAM;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(numDataNodes = 0, numClientNodes = 0, scope = ESIntegTestCase.Scope.TEST)
+public class ReindexPluginMetricsIT extends ESIntegTestCase {
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return Arrays.asList(ReindexPlugin.class, TestTelemetryPlugin.class);
+    }
+
+    protected ReindexRequestBuilder reindex() {
+        return new ReindexRequestBuilder(client());
+    }
+
+    protected UpdateByQueryRequestBuilder updateByQuery() {
+        return new UpdateByQueryRequestBuilder(client());
+    }
+
+    protected DeleteByQueryRequestBuilder deleteByQuery() {
+        return new DeleteByQueryRequestBuilder(client());
+    }
+
+    public static BulkIndexByScrollResponseMatcher matcher() {
+        return new BulkIndexByScrollResponseMatcher();
+    }
+
+    public void testReindexMetrics() throws Exception {
+        internalCluster().startNode();
+
+        indexRandom(
+            true,
+            prepareIndex("source").setId("1").setSource("foo", "a"),
+            prepareIndex("source").setId("2").setSource("foo", "a"),
+            prepareIndex("source").setId("3").setSource("foo", "b"),
+            prepareIndex("source").setId("4").setSource("foo", "c")
+        );
+        assertHitCount(prepareSearch("source").setSize(0), 4);
+
+        final String dataNodeName = internalCluster().getRandomNodeName();
+        final TestTelemetryPlugin testTelemetryPlugin = internalCluster().getInstance(PluginsService.class, dataNodeName)
+            .filterPlugins(TestTelemetryPlugin.class)
+            .findFirst()
+            .orElseThrow();
+
+        // Copy all the docs
+        ReindexRequestBuilder copy = reindex().source("source").destination("dest").refresh(true);
+        assertThat(copy.get(), matcher().created(4));
+        assertHitCount(prepareSearch("dest").setSize(0), 4);
+
+        // Use assertBusy to wait for all threads to complete so we get deterministic results
+        assertBusy(() -> {
+            testTelemetryPlugin.collect();
+            List<Measurement> measurements = testTelemetryPlugin.getLongHistogramMeasurement(REINDEX_TIME_HISTOGRAM);
+            assertThat(measurements.size(), equalTo(1));
+        });
+
+        // Now none of them
+        createIndex("none");
+        copy = reindex().source("source").destination("none").filter(termQuery("foo", "no_match")).refresh(true);
+        assertThat(copy.get(), matcher().created(0));
+        assertHitCount(prepareSearch("none").setSize(0), 0);
+        // wait for all threads to complete so that we get deterministic results
```

Comment:
can we remove this unused code?

Reply:
Yes! Sorry I wanted to keep it around until I made sure the assertBusy() was working :-). I did the ci build multiple times to make sure results are deterministic.